### PR TITLE
React Calculator

### PR DESCRIPTION
**Pull request by GraceOyiza**
- Add components directory.
- Add Display, App, Button and ButtonPanel components.

### CODE REVIEW

File: .eslintrc.json [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+  },
+  "extends": ["airbnb", "plugin:react/recommended"],
+  "plugins": ["react"],
+  "rules": {
+    "react/jsx-filename-extension": ["warn", { "extensions": [".js", ".jsx"] }],
+    "react/react-in-jsx-scope": "off",
+    "import/no-unresolved": "off",
+    "no-shadow": "off",
+    "arrow-parens": ["error", "as-needed"],
+    "linebreak-style": 0
```
Comment (bellom) [inline .eslintrc.json]: Oops, we have some little changes in this file, check out the [microverse-linter-config](https://github.com/microverseinc/linters-config/blob/master/react-redux/.eslintrc.json)
Kindly make use of the files provided by Microverse without editing them. However, if you're getting a linter error, do well to check the project requirement page under **TIP**

File: src/components/App.js [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+  return (
+    <>
+      <div className="App">
+        <div className="Appcalculator">
+          <Display />
+          <ButtonPannel>
+            {buttons.map(button => (
+              <Button key={button} name={button} />
+            ))}
+          </ButtonPannel>
```
Comment (bellom) [inline src/components/App.js]: Good job so far 👍 ....

But we need to move this mapping implementation inside the `buttonPanel.js`
image


image

**Review by bellom — CHANGES_REQUESTED** [review]
## Hi @GraceOyiza  :clap: 

### Awesome job so far!! :+1: 

#### You have done wonderfully well, you passed `almost` all the checklist for this project :+1: just a few issues left to work on............ :slightly_smiling_face: 

# Status: Changes Required! ♻️ 

**Kindly check out my comments below;**

### ♻️ Kindly re-submit for review after making all those changes :slightly_smiling_face: Thumps -up await you.....

**Best**

File: src/components/ButtonPanel.js [hunk 3]
```diff
@@ -0,0 +1,12 @@
+import PropTypes from 'prop-types';
+
+export default function ButtonPanel({ children }) {
+  return <div className="button_panel">{children}</div>;
```
Comment (bellom) [inline src/components/ButtonPanel.js]: Hint:

Microverse encourage DRY-(Don't Repeat Yourself) code, In your case; you repeated the `Button` component multiple times.

You can use [objects.key](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/keys)  and `map` to loop through the objects. `objects.keys` for the outer loop and `map` for the inner loop.

Example; 
```
const GroupButtons = {
  Group 1: ['AC', '+/-', '%', '÷'],
  Group 2: ['7', '8', '9', 'X'],
   ............
}
```

**Comment by GraceOyiza** [comment]
- Fix microverse lint errors without changing them.
- Move mapping implementation into ButtonPanel.js.
- Implement the DRY Principle

**Review by NtwaliHeritier — APPROVED** [review]
### **STATUS: APPROVED👍️👍️👍️👍️**

Hello @GraceOyiza 

Great work making the required changes👏️👏️👏️

You can merge👍️👍️

Happy coding💻️💻️

**Pull request merged**